### PR TITLE
feat(api): use `create_table` to load example data

### PR DESCRIPTION
**Pull request by jcrist**
This changes the loading of examples data to use `pyarrow.csv`/`pyarrow.parquet` to read the local data files, and `Backend.create_table(..., temp=True)` to create a table in the backend with this data for *most* backends. This lets our examples data work with most of our backends, making it easier to demo/try out ibis on a specific backend.

Some backends (duckdb & polars) will continue to read from the source files directly. We do this because these backends are local *and* support creating tables from views of source files (avoiding the need to store the data in memory). Since we have a few larger example datasets (e.g. IMDB) we want to use the `read_csv`/`read_parquet` methods for these backends to reduce memory usage, and make working with these larger example datasets smoother.

### CODE REVIEW

**Comment by jcrist** [comment]
This used to use `create_table` everywhere, but I ran into two issues:

- Memory usage would exceed the limits in CI when testing the larger datasets (IMDB, ...), since the `pyarrow -> create_table` path would have to read the data fully into memory. This would manifest as a SIGTERM in the test process, as github actions memory killer kicks in.
- Segfaults on windows only, also when working with the larger datasets. I haven't been able to distill out a reproducible example to send off to the duckdb team.

**Review by cpcloud — APPROVED** [review]
LGTM!

Few small comments, nothing blocking.

File: ibis/examples/__init__.py [hunk 1]
```diff
@@ -18,25 +19,79 @@
...
+                        quoted_strings_can_be_null=False,
+                    ),
+                )
+
+                # All null columns are inferred as null-type, but not all
+                # backends support null-type columns. Cast to an all-null
+                # string column instead.
+                for i, field in enumerate(table.schema):
+                    if pyarrow.types.is_null(field.type):
+                        table = table.set_column(i, field.name, table[i].cast("string"))
```
[the remark this answers is not in the document]
Comment (jcrist) [reply ibis/examples/__init__.py:77]: Unfortunately no, the schema provided to `.cast` needs to contain all the fields in the table, including those that aren't cast. I think the current 3-line implementation is clean enough.

File: ibis/expr/api.py [hunk 2]
```diff
@@ -850,26 +850,7 @@ def read_csv(sources: str | Path | Sequence[str | Path], **kwargs: Any) -> ir.Ta
...
-    │ allisdo01 │   1871 │     1 │ WS3    │ NA     │    27 │   133 │    28 │ … │
-    │ ansonca01 │   1871 │     1 │ RC1    │ NA     │    25 │   120 │    29 │ … │
-    │ armstbo01 │   1871 │     1 │ FW1    │ NA     │    12 │    49 │     9 │ … │
-    │ barkeal01 │   1871 │     1 │ RC1    │ NA     │     1 │     4 │     0 │ … │
-    │ barnero01 │   1871 │     1 │ BS1    │ NA     │    31 │   157 │    66 │ … │
-    │ barrebi01 │   1871 │     1 │ FW1    │ NA     │     1 │     5 │     1 │ … │
-    │ barrofr01 │   1871 │     1 │ BS1    │ NA     │    18 │    86 │    13 │ … │
-    │ …         │      … │     … │ …      │ …      │     … │     … │     … │ … │
-    └───────────┴────────┴───────┴────────┴────────┴───────┴───────┴───────┴───┘
+    >>> t = ibis.read_csv("path/to/data.csv")  # doctest: +SKIP
```
[the remark this answers is not in the document]
Comment (jcrist) [reply ibis/expr/api.py:853]: I'm going to leave this as is for now - I think the type annotation and  `Returns` section of the docstring is sufficient for letting user's know the return type.

**Pull request merged**